src/provider/default.ts

```diff
@@ -223,6 +224,27 @@ export class Provider implements ProviderInterface {
     return data;
   }
 
+  /**
+   * Gets the transaction receipt from a tx hash or tx id.
+   *
+   * [Reference] (https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/services/api/feeder_gateway/feeder_gateway_client.py#L104-L111)
+   *
+   * @param txHash
+   * @param txId
+   * @returns the transaction receipt object
+   */
+
+  public async getTransactionReceipt(
+    txHash?: BigNumberish,
+    txId?: BigNumberish
```

Comment:
I don't really like this. Should we limit it to txHash? With this interface it can be used as `getTransactionReceipt(undefined, 1)`, which I don't think is a great interface.

If we want to provide both options, maybe we should just have 2 different methods? wdyt

Reply:
Yes, I did not think about it and tried to keep it similar to Python implementation. But surely, it makes sense.

So we can have either have 2 different methods as you said or we can pass an object as an arguement. 

`getTransactionReceipt({ txHash, txId }: { txHash?: BigNumberish; txId?: BigNumberish });`

There after, we can use `getTransactionReceipt({txHash: someHash})`

Reply:
you could also try to detect hashes as they are much bigger than id's :)

Reply:
> Yes, I did not think about it and tried to keep it similar to Python implementation. But surely, it makes sense.
> 
> So we can have either have 2 different methods as you said or we can pass an object as an arguement.
> 
> `getTransactionReceipt({ txHash, txId }: { txHash?: BigNumberish; txId?: BigNumberish });`
> 
> There after, we can use `getTransactionReceipt({txHash: someHash})`

I think the solution I recommended is good, wdyt? Also we can take opinion of @0xs34n

Reply:
Passing an object sounds good to me. Thanks for the thoughtful discussions! @dhruvkelawala